src/ray/core_worker/core_worker.cc

```diff
@@ -1309,6 +1309,26 @@ Status CoreWorker::CreatePlacementGroup(
   return Status::OK();
 }
 
+Status CoreWorker::RemovePlacementGroup(const PlacementGroupID &placement_group_id) {
+  std::shared_ptr<std::promise<void>> promise = std::make_shared<std::promise<void>>();
+  std::shared_ptr<Status> removal_status = std::make_shared<Status>(Status::OK());
+  // Synchronously wait for placement group removal.
+  RAY_UNUSED(gcs_client_->PlacementGroups().AsyncRemovePlacementGroup(
+      placement_group_id, [promise, removal_status](Status status) {
+        *removal_status = status;
+        promise->set_value();
+      }));
+  if (promise->get_future().wait_for(std::chrono::seconds(
```

Comment:
Does this not block the main thread?

Reply:
It does. It is same for CreateActor and GetNamedActorHandle.

Reply:
I am doing this to avoid typical timing issues. For example, if remove is called twice in a row by mistake, it can cause unexpected behaviors. Another solution is just to make remove idempotent.